### PR TITLE
networkName and enableRhi configuration via NsxAlbInfraSetting for L4 VSes

### PR DESCRIPTION
**Pull request by shubhamavi**
This allows users to configure networkRef (in VirtualService) and enableRhi (in VsVip)
via the NsxAlbinfraSetting CRD.
This also adds upstream Gateway and GatewayClass CRD yamls into AKO's helm package
for installation via helm.

### CODE REVIEW

**Comment by anish-avi** [comment]
build ako

File: internal/rest/avi_obj_vs.go [hunk 1]
```diff
@@ -981,12 +997,25 @@ func (rest *RestOperations) AviVsVipCacheAdd(rest_op *utils.RestOp, vsKey avicac
 						continue
 					}
 					vsvipVips = append(vsvipVips, addr)
+					if ipamNetworkSubnet, ipamOk := vip["ipam_network_subnet"].(map[string]interface{}); ipamOk {
+						if networkRef, netRefOk := ipamNetworkSubnet["network_ref"].(string); netRefOk {
+							networkName = strings.Split(networkRef, "#")[1]
```
Comment (monotosh-avi) [inline internal/rest/avi_obj_vs.go:1002]: A length check here, and a couple of other places ?

**Review by monotosh-avi — COMMENTED** [review]
Looks good to me. @parimanur  can take a looks spl regarding public cloud.

File: internal/rest/avi_obj_vs.go [hunk 2]
```diff
@@ -767,7 +782,7 @@ func (rest *RestOperations) AviVsVipBuild(vsvip_meta *nodes.AviVSVIPNode, cache_
 		path = "/api/macro"
 		// Patch an existing vsvip if it exists in the cache but not associated with this VS.
 		vsvip_key := avicache.NamespaceName{Namespace: vsvip_meta.Tenant, Name: name}
-		utils.AviLog.Debugf("key: %s, searching in cache for vsVip Key: %s", key, vsvip_key)
+		utils.AviLog.Infof("key: %s, searching in cache for vsVip Key: %s", key, vsvip_key)
```
Comment (sudswasavi) [inline internal/rest/avi_obj_vs.go:785]: this can remain a debug log?

**Review by sudswasavi — COMMENTED** [review]
Just one comment. changes otherwise look good.

**Comment by shubhamavi** [comment]
build ako

**Review by monotosh-avi — APPROVED** [review]

**Review by sudswasavi — APPROVED** [review]

**Pull request merged**